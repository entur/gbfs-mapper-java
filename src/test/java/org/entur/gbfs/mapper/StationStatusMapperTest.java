package org.entur.gbfs.mapper;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import java.io.IOException;
import java.net.URL;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

@ExtendWith({SnapshotExtension.class})
class StationStatusMapperTest {
    private static final ObjectMapper objectMapper = new ObjectMapper();
    private Expect expect;

    @SnapshotName("gbfs_v2_3_to_v3_0_station_status_file_snapshot")
    @Test
    void testMapStationStatusFile() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v2_3/station_status.json");
        org.entur.gbfs.v2_3.station_status.GBFSStationStatus testSubject = objectMapper.readValue(resource, org.entur.gbfs.v2_3.station_status.GBFSStationStatus.class);
        org.entur.gbfs.v3_0_RC.station_status.GBFSStationStatus mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }

    @SnapshotName("gbfs_v3_0_to_v2_3_station_status_file_snapshot")
    @Test
    void testMapStationStatusFileInverse() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v3_0-RC/station_status.json");
        org.entur.gbfs.v3_0_RC.station_status.GBFSStationStatus testSubject = objectMapper.readValue(resource, org.entur.gbfs.v3_0_RC.station_status.GBFSStationStatus.class);
        org.entur.gbfs.v2_3.station_status.GBFSStationStatus mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }
}
