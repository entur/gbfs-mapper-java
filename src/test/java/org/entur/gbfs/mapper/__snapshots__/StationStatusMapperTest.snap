gbfs_v2_3_to_v3_0_station_status_file_snapshot=[
  {
    "last_updated": "2021-12-30T17:59:23.000+00:00",
    "ttl": 0,
    "version": "3.0",
    "data": {
      "stations": [
        {
          "station_id": "station1",
          "num_vehicles_available": 1,
          "vehicle_types_available": [
            {
              "vehicle_type_id": "abc123",
              "count": 1
            },
            {
              "vehicle_type_id": "def456",
              "count": 0
            }
          ],
          "num_vehicles_disabled": 2,
          "num_docks_available": 3,
          "num_docks_disabled": 1,
          "is_installed": true,
          "is_renting": true,
          "is_returning": true,
          "last_reported": "2021-01-05T17:02:05.000+00:00",
          "vehicle_docks_available": [
            {
              "vehicle_type_ids": [
                "abc123",
                "def456"
              ],
              "count": 2
            },
            {
              "vehicle_type_ids": [
                "def456"
              ],
              "count": 1
            }
          ]
        },
        {
          "station_id": "station2",
          "num_vehicles_available": 6,
          "vehicle_types_available": [
            {
              "vehicle_type_id": "abc123",
              "count": 2
            },
            {
              "vehicle_type_id": "def456",
              "count": 4
            }
          ],
          "num_vehicles_disabled": 1,
          "num_docks_available": 8,
          "num_docks_disabled": 1,
          "is_installed": true,
          "is_renting": true,
          "is_returning": true,
          "last_reported": "2021-01-05T17:01:46.000+00:00",
          "vehicle_docks_available": [
            {
              "vehicle_type_ids": [
                "abc123"
              ],
              "count": 6
            },
            {
              "vehicle_type_ids": [
                "def456"
              ],
              "count": 2
            }
          ]
        }
      ]
    }
  }
]


gbfs_v3_0_to_v2_3_station_status_file_snapshot=[
  {
    "last_updated": 1640887163,
    "ttl": 0,
    "version": "2.3",
    "data": {
      "stations": [
        {
          "station_id": "station1",
          "num_bikes_available": 1,
          "vehicle_types_available": [
            {
              "vehicle_type_id": "abc123",
              "count": 1
            },
            {
              "vehicle_type_id": "def456",
              "count": 0
            }
          ],
          "num_bikes_disabled": 2,
          "num_docks_available": 3,
          "num_docks_disabled": 1,
          "is_installed": true,
          "is_renting": true,
          "is_returning": true,
          "last_reported": 1609866125,
          "vehicle_docks_available": [
            {
              "vehicle_type_ids": [
                "abc123",
                "def456"
              ],
              "count": 2
            },
            {
              "vehicle_type_ids": [
                "def456"
              ],
              "count": 1
            }
          ]
        },
        {
          "station_id": "station2",
          "num_bikes_available": 6,
          "vehicle_types_available": [
            {
              "vehicle_type_id": "abc123",
              "count": 2
            },
            {
              "vehicle_type_id": "def456",
              "count": 4
            }
          ],
          "num_bikes_disabled": 1,
          "num_docks_available": 8,
          "num_docks_disabled": 1,
          "is_installed": true,
          "is_renting": true,
          "is_returning": true,
          "last_reported": 1609866106,
          "vehicle_docks_available": [
            {
              "vehicle_type_ids": [
                "abc123"
              ],
              "count": 6
            },
            {
              "vehicle_type_ids": [
                "def456"
              ],
              "count": 2
            }
          ]
        }
      ]
    }
  }
]